SU2_CFD/src/solvers/CGradientSmoothingSolver.cpp

```diff
@@ -309,8 +320,12 @@ void CGradientSmoothingSolver::ApplyGradientSmoothingDV(CGeometry *geometry, CNu
 
   /*--- Output the complete system matrix. ---*/
   if (rank == MASTER_NODE) {
-    ofstream SysMatrix(config->GetObjFunc_Hess_FileName());
+    /*--- For multizone append zone number to filename. ---*/
+    string hess_filename = config->GetObjFunc_Hess_FileName();
+    if (config->GetnZone() > 1) config->GetMultizone_FileName(hess_filename, config->GetiZone(), ".dat");
```

Comment:
Because we can also be in multizone mode with a single zone.
```suggestion
    if (config->GetMultizone()) config->GetMultizone_FileName(hess_filename, config->GetiZone(), ".dat");
```